subprojects/core/src/main/java/org/gradle/api/internal/project/taskfactory/AnnotationProcessingTaskFactory.java

```diff
@@ -52,16 +51,6 @@ public ITaskFactory createChild(ProjectInternal project, InstantiationScheme ins
     private <S extends Task> S process(S task) {
         TaskClassInfo taskClassInfo = taskClassInfoStore.getTaskClassInfo(task.getClass());
 
-        if (taskClassInfo.isIncremental()) {
```

Comment:
isn't this a business change already? Is this PR meant to do that?

Reply:
This change does not change the behaviour of the task, since we manually add the check in `DefaultTaskExecutionModeResolver`.

The only thing which changes for the users perspective is that `task.getHasOutput()` returns `false` in the incremental case. I don't think anybody relies on this (except for reporting).